- [ ] Update project with more microbes to improve the alignment
- [ ] Add todo list with updated task list
- [ ] Add project workflow
- [x] Add Makefile
- [x] Update the packages to the only ones useful to project
- [ ] Update environment.yml to have all the packages based on requirements.txt (use miniconda)
- [x] Add Dockerfile
- [x] Add License
- [ ] Add nine potential organisms to the 16S rRNA (*Neisseria gonorrhea*(https://www.ncbi.nlm.nih.gov/nuccore/X07714.1/ & paper normally streptomycin resistant https://www.ncbi.nlm.nih.gov/pmc/articles/PMC336862/?page=1), *Bordtella species*(https://www.ncbi.nlm.nih.gov/nuccore/AF177667 & https://www.ncbi.nlm.nih.gov/nuccore/AF177666 paper https://www.ncbi.nlm.nih.gov/pmc/articles/PMC86205/), *Mycobacterium tuberculosis* (search query "Mycobacterium tuberculosis"[Organism] AND 16S rRNA[All Fields]), *Salmonella Typhi*, *Mycoplasma genitalium*(potential https://www.ncbi.nlm.nih.gov/nuccore/X77334 and paper https://pubmed.ncbi.nlm.nih.gov/12517858/), *Clostridium difficile*(Clostridium difficile [Organism] AND 16S rRNA), *Pseudomonas aeruginosa* (Pseudomonas aeruginosa AND 16S rRNA) and Carbapenem-resistant Enterobacterales. Easier to go through search engine first to narrow down better
- [ ] Add CI/CD step
- [ ] Use biopython or other libraries to try automate steps of getting sequences
- [ ] Use [MEME](https://rosalind.info/glossary/meme/) to find motifs and improve clustering done before
- [ ] Add more different versions of python from 3.7, 3.8 and 3.9 in your tests. Just to circumvent the end of life python things to github actions.
- [ ] Update Dockerfile to Python 3.8-slim-buster
- [ ] Edit the requirements.txt on another branch for easy comparison.
